__tests__/propertyVolumeTests.ts

```diff
@@ -0,0 +1,21 @@
+import * as properties from '../src'
+import * as config from 'config'
+
+describe('Azure flex properties module should read from specified mount', () => {
+  test('should start have secrets read from volume', () => {
```

Comment:
can you add another test case which has default values and shows it correctly overrides it, and leaves alone properties it doesn't know about

Reply:
well there are no default values for the keyvaults ...
Are you talking about the mount point and prefix ?

Reply:
no the configuration values, 
i.e. 

given
/config/default.json
```
{
  "idam": {
    "api: "http://localhost:8080",
    "clientSecret": "123456"
  }
```
when
/mnt/keyVaults/idam/clientSecret
contains `654321`

then
`config.get('idam')`
```
{
    "api: "http://localhost:8080",
    "clientSecret": "654321"
  }
```

Reply:
it does not work like that though.
it will read the volumes
`/mnt/secrets/[keyvault]`
into properties with a `keyvaults` prefix
E.G
for the bulkscan s2s secret
config would look like this
```
keyvaults:
    s2s:
      microservicekey-bulk-scan-processor: "SOME_SECRET"
```

Reply:
so there is no default or properties that it will overwrite.

Reply:
users won't be reading from a keyvault locally so it needs to fit into that, locally defaults are set.
1. this is called properties volume and not keyvault, so it shouldn't be prefixed with that
2. whatever prefix is prepended will still need to be used locally, so a test should be written to prove it works

Reply:
sure done...
Took me a while to find the lodash merge.